anni-provider/src/providers/priority.rs

```diff
@@ -0,0 +1,157 @@
+use std::{borrow::Cow, collections::HashSet, num::NonZeroU8};
+
+use async_trait::async_trait;
+
+use crate::{AnniProvider, AudioResourceReader, ProviderError, Range, ResourceReader, Result};
+
+#[derive(Default)]
+pub struct PriorityProvider(Vec<(i32, Box<dyn AnniProvider + Send + Sync>)>);
```

Comment:
Maybe we can use [LinkedList](https://doc.rust-lang.org/std/collections/struct.LinkedList.html) here?

Reply:
Why? It would be harder to sort and possibly lower performance.